### DISCUSSION TITLE
How parse binary message

### DISCUSSION THREAD

**Discussion by Skr1pt** [SteamRE/SteamKit#1024] [General]
Hello, i am try parse binary answer to ClientMicroTxnAuthRequest, i get hex `150000240200FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEF22ED874801001001E28BDE6801004D6573736167654F626A65637400006C696E656974656D7300003000016465736372697074696F6E00E58F8DE68190E7B2BEE88BB13230E591A8E5B9B4E58DB0E88AB1E883B6E59B8A000767616D656974656D696400B14E0000000000000A616D6F756E74008C0A000000000000027175616E7469747900010000000808077472616E7369640016B9F2967FF79123074F726465724944008B7CCF440000000002617070696400DA0200000263757272656E637900120000000256415400000000000A746F74616C008C0A0000000000000A546178000000000000000000026C616E677561676500060000000273616E64626F7800000000000242696C6C696E6743757272656E637900120000000A42696C6C696E67546F74616C008C0A00000000000002537465616D5265616C6D0001000000025265717569726573436163686564506D744D6574686F64000000000002526566756E6461626C6500010000000808004D6573736167654F626A656374000248617357616C6C657400000000000808`
when i try decode it by c# methods i get `$ÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿï"í‡Hâ‹ÞhMessageObjectlineitems0descriptionåæç²¾è‹±20å‘¨å¹´å°èŠ±èƒ¶å›Šgameitemid±N
amountŒ
quantitytransid¹ò–÷‘#OrderID‹|ÏDappidÚcurrencyVAT
totalŒ

TaxlanguagesandboxBillingCurrency
BillingTotalŒ
SteamRealmRequiresCachedPmtMethodRefundableMessageObjectHasWallet`

**Comment by psychonic** [comment]
Have you tried deserializing it as a binary KV?

https://github.com/SteamRE/SteamKit/blob/b5d8502afa01385c7d4265f0a49dc3a56c03df23/SteamKit2/SteamKit2/Types/KeyValue.cs#L812

It looks like one, but there might be some other bytes before the KV structure starts. I don't know the exact structure of that message.

Edit: there are definitely bytes before the KV start that you'll need to skip.

**Comment by JustArchi** [comment]
As psychonic said above, you need to parse it to KV.

Example from ASF: https://github.com/JustArchiNET/ArchiSteamFarm/blob/fe43aa6cb6ac55c058cb19f121e1d58afcdbd2bd/ArchiSteamFarm/Steam/Integration/Callbacks/PurchaseResponseCallback.cs#L74-L82

**Reply by Skr1pt** [reply]
> As psychonic said above, you need to parse it to KV.
> 
> Example from ASF: https://github.com/JustArchiNET/ArchiSteamFarm/blob/fe43aa6cb6ac55c058cb19f121e1d58afcdbd2bd/ArchiSteamFarm/Steam/Integration/Callbacks/PurchaseResponseCallback.cs#L74-L82

yes, i try do this but all time failed. i think its must be like https://github.com/DoctorMcKay/node-binarykvparser,
i wanna do this action https://dev.doctormckay.com/topic/769-clientmicrotxnauthrequest-5504/
